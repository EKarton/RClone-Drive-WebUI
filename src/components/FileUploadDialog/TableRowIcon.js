import ArticleIcon from '@mui/icons-material/Article';
import DescriptionIcon from '@mui/icons-material/Description';
import FolderZipIcon from '@mui/icons-material/FolderZip';
import Tooltip from '@mui/material/Tooltip';

export default function TableRowIcon({ fileType }) {
  const renderIcon = () => {
    if (fileType === 'application/zip') {
      return <FolderZipIcon />;
    }

    if (fileType.startsWith('text')) {
      return <ArticleIcon />;
    }

    return <DescriptionIcon />;
  };

  return (
    <Tooltip title={fileType} arrow>
      <span>{renderIcon()}</span>
    </Tooltip>
  );
}
