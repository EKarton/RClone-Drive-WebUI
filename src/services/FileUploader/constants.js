export const ActionTypes = Object.freeze({
  UPLOAD_FILE: 'UPLOAD_FILE',
  CANCEL_UPLOAD: 'CANCEL_UPLOAD',
});
