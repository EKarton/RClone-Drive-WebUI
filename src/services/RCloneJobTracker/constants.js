export const ActionTypes = Object.freeze({
  TRACK_NEW_JOB: 'TRACK_NEW_JOB',
  UNTRACK_NEW_JOB: 'UNTRACK_NEW_JOB',
});
