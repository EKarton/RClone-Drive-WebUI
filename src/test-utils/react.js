import { FileViewerProvider } from 'contexts/FileViewerStore';
import { RCloneInfoStateProvider } from 'contexts/RCloneInfoStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState, initialRCloneInfoState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  const renderedComponent = render(
    <RCloneInfoStateProvider defaultState={initialRCloneInfoState}>
      <FileViewerProvider defaultState={initialFileViewerState}>
        <Router history={history}>{component}</Router>
      </FileViewerProvider>
    </RCloneInfoStateProvider>
  );

  return {
    ...renderedComponent,
    history,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
