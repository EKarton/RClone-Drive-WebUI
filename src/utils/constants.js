export const ICON_SIZE = Object.freeze({
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
});

export const StatusTypes = Object.freeze({
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
});

export const JobStatus = Object.freeze({
  ONGOING: 'ONGOING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
});

export const JobTypes = Object.freeze({
  UPLOAD_FILE: 'UPLOAD_FILE',
  MOVE_FILE: 'MOVE_FILE',
  MOVE_FOLDER: 'MOVE_FOLDER',
  RENAME_FILE: 'RENAME_FILE',
  RENAME_FOLDER: 'RENAME_FOLDER',
});

export const COLOR_MODE = Object.freeze({
  LIGHT: 'light',
  DARK: 'dark',
});

export const ImageMimeTypes = new Set(['image/jpeg', 'image/png']);
