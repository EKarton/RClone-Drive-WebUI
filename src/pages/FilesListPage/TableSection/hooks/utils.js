import { JobStatus, JobTypes } from 'utils/constants';

export function getFileNames(data) {
  return new Set(data.filter((file) => !file.IsDir).map((file) => file.Name));
}

export function getFolderNames(data) {
  return new Set(data.filter((file) => file.IsDir).map((file) => file.Name));
}

export function getUploadingJobs(jobs, remote, path) {
  return jobs.filter(
    (job) =>
      job.jobType === JobTypes.UPLOAD_FILE &&
      job.remote === remote &&
      job.dirPath.startsWith(path) &&
      job.status.value !== JobStatus.SUCCESS
  );
}

export function getUploadingFiles(jobs, remote, path) {
  return getUploadingJobs(jobs, remote, path).filter((job) => job.dirPath === path);
}

export function getUploadingFolders(jobs, remote, path) {
  return getUploadingJobs(jobs, remote, path).filter((job) => job.dirPath !== path);
}

export function getMovingJobs(jobs, remote, path) {
  const supportedJobTypes = new Set([
    JobTypes.MOVE_FILE,
    JobTypes.MOVE_FOLDER,
    JobTypes.RENAME_FILE,
    JobTypes.RENAME_FOLDER,
  ]);

  return jobs.filter((job) => {
    const correctJob = supportedJobTypes.has(job.jobType);
    const inProgress = job.status.value === JobStatus.ONGOING;

    if (!correctJob || !inProgress) {
      return false;
    }

    const isMatchSrc = job.src?.remote === remote && job.src?.dirPath === path;
    const isMatchTarget = job.target?.remote === remote && job.target?.dirPath === path;

    return correctJob && inProgress && (isMatchSrc || isMatchTarget);
  });
}
