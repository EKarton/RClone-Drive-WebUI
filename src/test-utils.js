import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
