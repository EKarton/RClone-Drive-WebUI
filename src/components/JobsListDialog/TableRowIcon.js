import BackupIcon from '@mui/icons-material/Backup';
import DriveFileMoveIcon from '@mui/icons-material/DriveFileMove';
import DriveFileRenameOutlineIcon from '@mui/icons-material/DriveFileRenameOutline';
import TaskIcon from '@mui/icons-material/Task';
import Tooltip from '@mui/material/Tooltip';
import PropTypes from 'prop-types';
import { JobTypes } from 'utils/constants';

export default function TableRowIcon({ jobType }) {
  const renderIcon = () => {
    switch (jobType) {
      case JobTypes.UPLOAD_FILE: {
        return <BackupIcon />;
      }
      case JobTypes.MOVE_FILE:
      case JobTypes.MOVE_FOLDER: {
        return <DriveFileMoveIcon />;
      }
      case JobTypes.RENAME_FILE:
      case JobTypes.RENAME_FOLDER: {
        return <DriveFileRenameOutlineIcon />;
      }
      default: {
        return <TaskIcon />;
      }
    }
  };

  return (
    <Tooltip title={jobType} arrow>
      <span>{renderIcon()}</span>
    </Tooltip>
  );
}

TableRowIcon.propTypes = {
  jobType: PropTypes.string,
};
