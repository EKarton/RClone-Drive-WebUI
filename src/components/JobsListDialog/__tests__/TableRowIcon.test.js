import { JobTypes } from 'utils/constants';
import { render } from 'test-utils/react';
import TableRowIcon from '../TableRowIcon';

describe('TableRowIcon', () => {
  it.each([
    JobTypes.MOVE_FILE,
    JobTypes.MOVE_FOLDER,
    JobTypes.RENAME_FILE,
    JobTypes.RENAME_FOLDER,
    JobTypes.UPLOAD_FILE,
    'Random Job',
  ])('should match snapshot given job type is %s', (jobType) => {
    const { baseElement } = render(<TableRowIcon jobType={jobType} />);

    expect(baseElement).toMatchSnapshot();
  });
});
