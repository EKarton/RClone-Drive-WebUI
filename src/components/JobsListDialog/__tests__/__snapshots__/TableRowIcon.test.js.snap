// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRowIcon should match snapshot given job type is MOVE_FILE 1`] = `
<body>
  <div>
    <span
      aria-label="MOVE_FILE"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DriveFileMoveIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6h-8l-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-6 12v-3h-4v-4h4V8l5 5-5 5z"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`TableRowIcon should match snapshot given job type is MOVE_FOLDER 1`] = `
<body>
  <div>
    <span
      aria-label="MOVE_FOLDER"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DriveFileMoveIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6h-8l-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-6 12v-3h-4v-4h4V8l5 5-5 5z"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`TableRowIcon should match snapshot given job type is RENAME_FILE 1`] = `
<body>
  <div>
    <span
      aria-label="RENAME_FILE"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DriveFileRenameOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.41 5.8 17.2 4.59c-.78-.78-2.05-.78-2.83 0l-2.68 2.68L3 15.96V20h4.04l8.74-8.74 2.63-2.63c.79-.78.79-2.05 0-2.83zM6.21 18H5v-1.21l8.66-8.66 1.21 1.21L6.21 18zM11 20l4-4h6v4H11z"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`TableRowIcon should match snapshot given job type is RENAME_FOLDER 1`] = `
<body>
  <div>
    <span
      aria-label="RENAME_FOLDER"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DriveFileRenameOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.41 5.8 17.2 4.59c-.78-.78-2.05-.78-2.83 0l-2.68 2.68L3 15.96V20h4.04l8.74-8.74 2.63-2.63c.79-.78.79-2.05 0-2.83zM6.21 18H5v-1.21l8.66-8.66 1.21 1.21L6.21 18zM11 20l4-4h6v4H11z"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`TableRowIcon should match snapshot given job type is Random Job 1`] = `
<body>
  <div>
    <span
      aria-label="Random Job"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="TaskIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm-3.06 16L7.4 14.46l1.41-1.41 2.12 2.12 4.24-4.24 1.41 1.41L10.94 18zM13 9V3.5L18.5 9H13z"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`TableRowIcon should match snapshot given job type is UPLOAD_FILE 1`] = `
<body>
  <div>
    <span
      aria-label="UPLOAD_FILE"
      class=""
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="BackupIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
        />
      </svg>
    </span>
  </div>
</body>
`;
